apis/v1beta1/shared_types.go

```diff
@@ -480,6 +480,20 @@ type AnnotationKey string
 type AnnotationValue string
 
 // AddressType defines how a network address is represented as a text string.
+// This may take two possible forms:
+//
+// * A predefined CamelCase string identifier (currently limited to `IPAddress` or `Hostname`)
```

Comment:
Just because these types of references with point-in-time statements of facts about other symbols in the code run the risk of bit-rotting and being missed when other things change, would it be worth making this change?

```suggestion
// * A predefined CamelCase string identifier (e.g. `IPAddress`, `Hostname`)
```

Reply:
I definitely agree that it's a risk that they get out of sync, but I prefer this to the alternative of requiring users to parse a difficult to understand regex to understand which values are currently valid. We do need to be very careful with all reviews around enums like this that when we change the set of allowed values, we update that _everywhere_.